src/utils/slurmd.py

```diff
@@ -0,0 +1,136 @@
+#!/usr/bin/python3
+# Copyright 2023 Canonical Ltd.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Manage the internal slurmd daemon on Juju machines.
+
+This module also provides a wrapper for starting the slurmd service using systemd.
+"""
+
+import datetime
+import logging
+import os
+import shlex
+import subprocess
+import sys
+import textwrap
+from pathlib import Path
+
+import charms.operator_libs_linux.v1.systemd as systemd
+
+_logger = logging.getLogger(__name__)
+
+
+def start() -> None:
+    """Start slurmd service."""
+    systemd.service_start("slurmd")
+
+
+def stop() -> None:
+    """Stop slurmd service."""
+    systemd.service_stop("slurmd")
+
+
+def restart() -> None:
+    """Restart slurmd service."""
+    stop()
```

Comment:
Is there any reason a systemd.service_restart("slurmd") is not used here? A stop and start is not a problem, but should probably leave a comment explaining the why if the service_restart doesn't apply or someone like me will come along and say "oh, we should make this a service_restart"

Reply:
To be honest, I just wrote it like this because I had already defined the start/stop methods, and I had originally only imported the service_start/service_stop methods rather than the whole systemd charmlib namespace. There is no benefit to using start/stop v.s. service_restart, so I can just switch it to use `systemd.service_restart(...)`.

Reply:
Refactored in commit https://github.com/omnivector-solutions/slurmd-operator/pull/21/commits/a4c776dabbe3d64f516c45df5bbae0e2041d1bff